modules/networking/main.tf

```diff
@@ -0,0 +1,75 @@
+resource "google_compute_network" "vpc" {
+  name                    = "${var.prefix}-network"
+  auto_create_subnetworks = false
+  project                 = var.project_id
+
+  lifecycle {
+    create_before_destroy = true
+    prevent_destroy       = false
+  }
+
+}
+
+resource "google_compute_route" "default_route" {
+  name             = "${var.prefix}-default-route"
+  project          = var.project_id
+  network          = google_compute_network.vpc.name
+  dest_range       = "0.0.0.0/0"
+  priority         = 1000
+  next_hop_gateway = "default-internet-gateway"
+
+  # Ensure this is destroyed before the network
+  depends_on = [google_compute_network.vpc]
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "google_compute_subnetwork" "subnet" {
+  name          = "${var.prefix}-subnet"
+  project       = var.project_id
+  network       = google_compute_network.vpc.id
+  ip_cidr_range = var.subnet_cidr
+  region        = var.region
+
+  private_ip_google_access = true
+
+  secondary_ip_range {
+    range_name    = "pods"
+    ip_cidr_range = var.pods_cidr
+  }
+
+  secondary_ip_range {
+    range_name    = "services"
+    ip_cidr_range = var.services_cidr
+  }
+
+}
+
+resource "google_compute_global_address" "private_ip_address" {
+  provider      = google
+  project       = var.project_id
+  name          = "${var.prefix}-private-ip"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = google_compute_network.vpc.id
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "google_service_networking_connection" "private_vpc_connection" {
+  provider                = google
+  network                 = google_compute_network.vpc.id
+  service                 = "servicenetworking.googleapis.com"
+  reserved_peering_ranges = [google_compute_global_address.private_ip_address.name]
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  deletion_policy = "ABANDON"
```

Comment:
Why do we abandon here?

Reply:
We initially added that as we’ve hit an issue in CI where the VPC peering gets deleted before its dependent resources (like Cloud SQL or GKE), which causes the destroy to fail.